pystorm/component.py

```diff
@@ -498,13 +498,22 @@ def run(self):
                 except StormWentAwayError:
                     log.error(log_msg, exc_info=exc_info)
                     log.info('Exiting because parent Storm process went away.')
-                    sys.exit(2)
+                    self._exit(2)
                 except:
                     log.error(log_msg, exc_info=exc_info)
                     log.error('While trying to handle previous exception...',
                               exc_info=sys.exc_info())
+
                 if self.exit_on_exception:
-                    sys.exit(1)
+                    self._exit(1)
+
+    def _exit(self, status_code):
+        """Properly kill Python process including zombie threads."""
+        # If there are active threads still running infinite loops, sys.exit
+        # won't kill them but os._exit will. os._exit skips calling cleanup
```

Comment:
We're going to need this documented somewhere so that someone isn't relying on these cleanup actions to occur. At the very least, it needs to be in the release notes in flashing letters.